core/src/main/java/com/linecorp/armeria/common/unsafe/ByteBufHttpData.java

```diff
@@ -169,11 +181,22 @@ public String toString() {
 
     @Override
     public InputStream toInputStream() {
-        return new ByteBufInputStream(buf.retainedDuplicate(), true);
+        return new ByteBufInputStream(buf.duplicate(), false);
     }
 
     @Override
-    public ByteBufHttpData withEndOfStream() {
-        return new ByteBufHttpData(buf, true);
+    public ByteBufHttpData withEndOfStream(boolean endOfStream) {
+        if (endOfStream == this.endOfStream) {
+            return this;
+        }
+        return new ByteBufHttpData(buf, endOfStream);
+    }
```

Comment:
This is not normal behavior, however a `IllegalReferenceCountException` could be raised by:
```java
final ByteBuf byteBuf = Unpooled.wrappedBuffer("hello".getBytes());
final PooledHttpData pooledHttpData1 = PooledHttpData.wrap(byteBuf);
pooledHttpData1.close();
final PooledHttpData polledHttpData2 = pooledHttpData1.withEndOfStream(true);
// io.netty.util.IllegalReferenceCountException: refCnt: 0, decrement: 1
polledHttpData2.close();
```
This method does not copy the `ByteBufHttpData.closed`.

Reply:
Yup good idea